File: 스택_큐/[영우] 스택_큐/올바른_괄호.js
/**
 * 아이디어
 * 각 문자열을 순회하면서 열리고 닫힌 걸 count 로 센다
 * 1. ( 가 나오면 count++
 * 2. ) 가 나오면 count--
 * 3. count 가 0보다 작으면 false
 * 4. count 가 0이면 true
 * 5. count 가 0보다 크면 false
 */

function solution(s) {
  let count = 0;

  for (let i = 0; i < s.length; i++) {
    if (s[i] === "(") {
      count++;
    } else {
      count--;
    }

    if (count < 0) {
      return false;
    }
  }

  return count === 0 ? true : false;
}

File: 스택_큐/[영선] 스택_큐/올바른 괄호.swift
//
//  올바른 괄호.swift
//  Algorithm
//
//  Created by 임영선 on 2023/11/03.
//

import Foundation

func solution(_ s: String) -> Bool{
    var stack = ""
    
    for i in s{
        if(i == "("){
            stack.append("(")
        }else{
            if(stack.isEmpty){
                return false
            }else{
                stack.popLast()
            }
        }
    }
    return stack.isEmpty ? true : false
}
